### PR TITLE
fix: Insert data guide

### LINKED ISSUE #1901: [Docs]: Guide to insert data using Form
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Documentation Link

_No response_

### Discord/slack/intercom Link

_No response_

### Describe the problem and improvement.

Write a how-to guide to insert data using a Form widget.

### PR DESCRIPTION
**Pull request by rashmi-sahoo-git**
closes #1901 
## Checklist
I have:
- [ ] run the content through Grammarly
- [ ] linked to sample apps when relevant
- [ ] added the meta description for each page in the PR
- [ ] minimized the callouts and added only when necessary
- [ ] added the `queryString` parameter to the Tabs (if used)
- [ ] masked PII in images. For example, login credentials, account details, and more
- [ ] added images only when necessary
- [ ] deleted the images that are no longer used for the updated pages in the PR
- [ ] followed the image file naming convention while renaming or adding new images. (Use lowercase letters, dashes between words, and be as descriptive as possible)
- [ ] used the `<figure/>` tag instead of a markdown representation for images
- [ ] added the `<figcaption/>` tag to add a caption to the image
- [ ] added the `alt` attribute in the `` tag
- [ ] verified and updated the cross-references or created redirect rules for the changed or removed content 
- [ ] reviewed and applied the style changes for UI formatting. For example, Bold the UI elements(Buttons on screen) used in the doc.

### CODE REVIEW

File: website/docs/build-apps/how-to-guides/insert-data.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+1. Add a Button widget. Set the widget's **onClick** event to open a new page by selecting the **Navigate to** action and selecting the appropriate page.
+   For more information, see [navigateTo](/reference/appsmith-framework/widget-actions/navigate-to).
+2. In the new page, create a Form widget to capture the details you want to insert.
+3. You can update fields dynamically by setting the events of the widgets to bind data to other fields in the Form.
+   For example, to set the `offerCode` based on the selection of `location` for a new offer use the following code:
+
+   ```jsx
+   {{ offerCode.setText(location.selectedOptionValue + code) }}
+   ```
+   For more information, see [Dynamic data mapping](/reference/widgets/list#dynamic-data-mapping).
```
Comment (danciaclara) [inline website/docs/build-apps/how-to-guides/insert-data.md:20]:
```suggestion
   For more information, see [Bind Data to Wdigets](/core-concepts/building-ui/dynamic-ui).
```

File: website/docs/build-apps/how-to-guides/insert-data.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+This guide shows you how to insert data into a datasource using the Form widget in a new page and a Modal. It explains steps to bind data to widgets, validate user input, and configure query.
+
+## Using new page
+To insert data in a new page, follow these steps:
+1. Add a Button widget. Set the widget's **onClick** event to open a new page by selecting the **Navigate to** action and selecting the appropriate page.
+   For more information, see [navigateTo](/reference/appsmith-framework/widget-actions/navigate-to).
+2. In the new page, create a Form widget to capture the details you want to insert.
+3. You can update fields dynamically by setting the events of the widgets to bind data to other fields in the Form.
+   For example, to set the `offerCode` based on the selection of `location` for a new offer use the following code:
```
Comment (danciaclara) [inline website/docs/build-apps/how-to-guides/insert-data.md:15]: Just write example. Below the example, give them context of what is the use of that code

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/sidebars.js [hunk 3]
```diff
@@ -331,6 +331,8 @@ const sidebars = {
           link: { type: 'doc', id: 'build-apps/how-to-guides/README' },
           items: [
             'core-concepts/building-ui/dynamic-ui/README',
+            'build-apps/how-to-guides/insert-data',
+            'build-apps/how-to-guides/submit-form-data',
             {
               type: 'category',
               label: 'Display and Lookup Table Data',
```
[the remark this answers is not in the document]
Comment (rashmi-sahoo-git) [reply website/sidebars.js]: Done

File: website/docs/build-apps/how-to-guides/insert-data.md (hunk 2 above)
Comment (rashmi-sahoo-git) [reply website/docs/build-apps/how-to-guides/insert-data.md]: Done

**Review by danciaclara — APPROVED** [review]

**Pull request merged**